static/app/views/dashboards/widgetQueryQueue.tsx

```diff
@@ -0,0 +1,94 @@
+// scaffold basic context provider and hook
+
+import {createContext, useCallback, useContext, useRef} from 'react';
+import {metrics} from '@sentry/react';
+import {
+  asyncQueuerOptions,
+  useAsyncQueuer,
+  type ReactAsyncQueuer,
+} from '@tanstack/react-pacer';
+
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
+import type GenericWidgetQueries from 'sentry/views/dashboards/widgetCard/genericWidgetQueries';
+
+type QueueItem<SeriesResponse, TableResponse> = {
+  widget: GenericWidgetQueries<SeriesResponse, TableResponse>;
+};
+
+export type WidgetQueryQueue = ReactAsyncQueuer<QueueItem<any, any>>;
```

Comment:
we can use `unknown` instead of `any`:


```suggestion
export type WidgetQueryQueue = ReactAsyncQueuer<QueueItem<unknown, unknown>>;
```